edx/frontend-app-discussions
# [BD-03] [BB-2899] Embed routes for discussions MFE

Provides Embed routes for the Discussions, Topics, and Posts elements. All routing has been rebuilt in this PR. 

Only the last commit (https://github.com/edx/frontend-app-discussions/pull/4/commits/c8ff51a550049ee6749fa30e8473782154fbcb85) contains code relevant to this PR.

**JIRA tickets**:
[BD-3](https://openedx.atlassian.net/browse/BD-3)
[BLENDED-571](https://openedx.atlassian.net/browse/BLENDED-571)

**Dependencies**: 

This PR is directly based off https://github.com/edx/frontend-app-discussions/pull/2

**Testing instructions**:

1. Go to your LMS, login as the `edx` user, and enroll in a course (any course) and create a post in the discussion forum. Repeat with at least 1 other user. Ensure one post contains replies, and there is at least 1 discussion type post, and 1 question type post.
2. Copy the course key for the course you made comments in
3. Launch the frontend-app-discussions application with `npm install && npm start`
4. Navigate to `locahost:2002/COURSE_KEY/embed/discussion/topics`. This is the Discussion Embedded UI. Note that the  header and footer are now gone
5. Play with the application in this view, noting that whenever a link is created, the URL always contains `/embed/discussion`.
6. Click **All Topics**, and select the topic group (with the gray headers) containing the topic you commented in
7. Modify the URL, replacing `/embed/discussion` with `/embed/topic`. This is the Topic Embedded UI. Note that the Breadcrumb, Navigation Bar and Search Bar are now all gone
8. Play with the application in this view, noting that whenever a link is created, the URL always contains `/embed/topic` and the navigation is limited without modifications to the URL.
9. Navigate to any post.
10. Modify the URL, replacing `/embed/topic` with `/embed/post`. This is the Post Embedded UI
11. Note that only the post and it's comments are visible with no way to navigate to other posts

**Authors Concerns**:

Currently, when embedding a Topic Group using the Embedded Topic UI, once you select a topic, you cannot navigate back to the group without hitting _back_ on your browser.. I feel like rendering a Breadcrumb in that view, even if only for the Topic Group embeds, might be beneficial so that users can at least navigate back to the group.

**Reviewers**
